core/src/main/java/org/apache/iceberg/avro/SchemaToType.java

```diff
@@ -72,7 +72,7 @@ private int getId(Schema.Field field) {
     }
   }
 
-  private int allocateId() {
+  protected int allocateId() {
```

Comment:
Can we pass in a function instead, like we do in `TypeUtil`? https://github.com/apache/incubator-iceberg/blob/master/api/src/main/java/org/apache/iceberg/types/TypeUtil.java#L185-L190

Reply:
If this seems generally useful. I can create a ticket for this